src/main/java/com/Repository/EventRepository.java

```diff
@@ -1,17 +1,11 @@
 package com.Repository;
 
 import com.Model.Event;
-import org.springframework.data.domain.Page;
-import org.springframework.data.domain.Pageable;
 import org.springframework.data.repository.PagingAndSortingRepository;
-import org.springframework.stereotype.Component;
-import org.springframework.stereotype.Repository;
-
 import java.util.Date;
-import java.util.List;
 
 public interface EventRepository extends PagingAndSortingRepository<Event, Integer> {
 
-    public List<Event> findAllByNameAndDate(String name, Date date);
+    public Event findAllByNameAndDate(String name, Date date);
```

Comment:
shouldn't findAll... return a list/Iterable of Events? Either the method name or the signature is misleading.

Reply:
do you have a test to see if this works as expected? I doubt that it does.

Reply:
I changed it to findByNameAndDate(String name, Date date).  
Test cases added inside EventTest.java